airbyte-ci/connectors/connector_ops/connector_ops/utils.py

```diff
@@ -555,6 +556,20 @@ def normalization_tag(self) -> Optional[str]:
     def is_using_poetry(self) -> bool:
         return Path(self.code_directory / "pyproject.toml").exists()
 
+    @property
+    def is_released(self) -> bool:
+        metadata = self.metadata
+        in_registry = False
+        registry = download_catalog(OSS_CATALOG_URL)
+        for connector in registry[f"{self.connector_type}s"]:
+            if (
+                connector[f"{self.connector_type}DefinitionId"] == metadata["definitionId"]
```

Comment:
Wait a second, does each version of a connector have a different definition ID? Because we want to run all checks on each new version in case new version screws things up.

Reply:
Definitions IDs are static and are connector's primary key. The logic here is:
If the definition ID and the dockerImage version are found in the registry it means the connector was published.